### PR TITLE
Fix ETH class

### PR DESCRIPTION
**Pull request by GQM**
Made small correction to ETH class due a slight different connection mechanism. Also added dynamic variables to pull from currencies.yml file

### CODE REVIEW

**Comment by muhammednagy** [comment]
Hello
Let's test that first?

**Comment by bitcoinli** [comment]
does it work ?

**Comment by muhammednagy** [comment]
didn't test it yet

**Comment by atlanticstarr1** [comment]
Can we upgrade to Ruby > 2.2.1 ? Getting issues with bundler for ruby version 2.2.1

**Comment by muhammednagy** [comment]
can u describe the problems?

**Comment by ParthBarot-BoTreeConsulting** [comment]
@muhammednagy So what is the next step? As I see, the PR is not yet merged into https://github.com/InfraexDev/peatio and without this, I guess it would not work for `ETH` as it would need the proper URL. 

Thanks.

File: app/models/worker/deposit_coin_address.rb [hunk 1]
```diff
@@ -8,11 +8,13 @@ def process(payload, metadata, delivery_info)
       return if payment_address.address.present?
 
       currency = payload[:currency]
+      c = Currency.find_by_code(currency.to_s)
+
       if currency == 'eth'
         address  = CoinRPC[currency].personal_newAccount("")
-        open('http://your_eth_server_ip/cgi-bin/restart.cgi')
+        open(c.rpc + '/cgi-bin/restart.cgi')
```
Comment (ParthBarot-BoTreeConsulting) [inline app/models/worker/deposit_coin_address.rb]: @muhammednagy Why there is a need to call this URL? Is it necessary to reboot after we create a new account using `CoinRPC['eth'].personal_newAccount("")` ?

Comment (muhammednagy) [reply app/models/worker/deposit_coin_address.rb]: i will remove this line 
it was necessary 
it isn't now

**Comment by muhammednagy** [comment]
that's not true 
you just need to update the URL manually

**Comment by frezeq0** [comment]
bittrex opensource: https://github.com/frezeq0/bittrex-opensource

**Pull request closed without merging**